File: push/src/main/java/com/xiaomi/push/service/SendMessageJob.java
package com.xiaomi.push.service;

import com.xiaomi.channel.commonutils.logger.MyLog;
import com.xiaomi.push.service.XMPushService;
import com.xiaomi.slim.Blob;
import com.xiaomi.smack.XMPPException;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/service/SendMessageJob.class */
class SendMessageJob extends XMPushService.Job {
    private Blob mBlob;
    private XMPushService pushService;

    public SendMessageJob(XMPushService xMPushService, Blob blob) {
        super(4);
        this.pushService = null;
        this.pushService = xMPushService;
        this.mBlob = blob;
    }

    @Override // com.xiaomi.push.service.XMPushService.Job
    public String getDesc() {
        return "send a message.";
    }

    @Override // com.xiaomi.push.service.XMPushService.Job
    public void process() {
        try {
            Blob blob = this.mBlob;
            if (blob != null) {
                this.pushService.sendPacket(blob);
            }
        } catch (XMPPException e) {
            MyLog.e(e);
            this.pushService.disconnect(10, e);
        }
    }
}

File: push/src/main/java/com/xiaomi/push/service/SendMessageJob.kt
package com.xiaomi.push.service

import com.xiaomi.channel.commonutils.logger.MyLog
import com.xiaomi.slim.Blob
import com.xiaomi.smack.XMPPException

class SendMessageJob(
    private val pushService: XMPushService,
    private val blob: Blob?,
) : XMPushService.Job(4) {
    override fun getDesc(): String = "send a message."

    override fun process() {
        try {
            blob?.let(pushService::sendPacket)
        } catch (e: XMPPException) {
            MyLog.e(e)
            pushService.disconnect(10, e)
        }
    }
}
